src/main/kotlin/software/amazon/smithy/intellij/index/SmithyBuildConfigurationIndex.kt

```diff
@@ -47,9 +47,17 @@ class SmithyBuildConfigurationIndex : SingleEntryFileBasedIndexExtension<SmithyB
     override fun getValueExternalizer(): DataExternalizer<SmithyBuildConfiguration> =
         object : DataExternalizer<SmithyBuildConfiguration> {
             override fun save(out: DataOutput, config: SmithyBuildConfiguration) =
-                out.writeUTF(SmithyJson.writeValueAsString(config))
+                out.writeBytes(SmithyJson.writeValueAsString(config))
```

Comment:
```
override fun save(out: DataOutput, config: SmithyBuildConfiguration) {
    val bytes = SmithyJson.writeValueAsBytes(config)
    out.write(bytes.size)
    out.write(bytes)
}
```

would simplify the read logic a bit

```
override fun read(`in`: DataInput): SmithyBuildConfiguration {
    val bytes = ByteArray(`in`.readInt())
    `in`.readFully(bytes)
    return SmithyJson.readValue<SmithyBuildConfiguration>(bytes)
}
```

Reply:
Good catch. :) Was moving too quickly, saw that `write` took in an `int` and went with it. :P